src/base/bittorrent/torrenthandle.h

```diff
@@ -445,14 +445,26 @@ namespace BitTorrent
 
         QHash<QString, TrackerInfo> m_trackerInfos;
 
-        enum StartupState
+        enum ProcessingState
         {
-            NotStarted,
-            Starting,
-            Started
+            PreparingPaused, // torrent is preparing to start processing in "paused" state
+            PreparingResumed, // torrent is preparing to start processing in "resumed" state
+            Starting, // torrent is prepared and starting to perform regular processing
+            Working, // torrent is performing regular processing
+            CheckingPaused // torrent is checking in "paused" state
```

Comment:
Aren't torrent checking is always done in paused state? If so then it could be named just `Checking`.

Reply:
User can perform Checking action on both Paused and Resumed torrent. But to check it actually we need to resume paused torrent temporarily. It's required to handle this extra state in a special way (e.g. if user close qBittorrent during checking all affected torrents should be restored in its original state but current master restores them all resumed).